src/Razor/src/Microsoft.AspNetCore.Razor.Language/Components/ComponentBindLoweringPass.cs

```diff
@@ -609,62 +614,65 @@ private bool TryParseBindAttribute(BindEntry bindEntry, out string valueAttribut
             });
         }
 
-        private void RewriteNodesForEventCallbackBind(
+        private void RewriteNodesForComponentEventCallbackBind(
+            IntermediateToken original,
+            List<IntermediateToken> valueExpressionTokens,
+            List<IntermediateToken> changeExpressionTokens)
+        {
+            valueExpressionTokens.Add(original);
+
+            changeExpressionTokens.Add(new IntermediateToken()
+            {
+                Content = $"{ComponentsApi.RuntimeHelpers.CreateInferredEventCallback}(this, __value => {original.Content} = __value, {original.Content})",
+                Kind = TokenKind.CSharp
+            });
+        }
+
+        private void RewriteNodesForElementEventCallbackBind(
             IntermediateToken original,
             IntermediateToken format,
             IntermediateToken culture,
-            BoundAttributeDescriptor valueAttribute,
-            BoundAttributeDescriptor changeAttribute,
             List<IntermediateToken> valueExpressionTokens,
             List<IntermediateToken> changeExpressionTokens)
         {
-            if (changeAttribute == null)
```

Comment:
We were using this check to test "is this a component or not" - which isn't really correct. Components can support bind to arbitrary parameters (you can do `@bind-Message` even if `Message` isn't defined).